judge/runguard.c

```diff
@@ -999,8 +1002,9 @@ int main(int argc, char **argv)
 			break;
 		case 'u': /* user option: uid or string */
 			use_user = 1;
-			runuid = strtol(optarg,&ptr,10);
 			runuser = strdup(optarg);
```

Comment:
I guess we also would want some error checking on strdup returning null and its errno?

Reply:
added